R/changeLegendColour.R

```diff
@@ -23,38 +23,32 @@ changeLegendColour <- function(x,
   ##Build plot to extract deets from it
   g <- ggplot2::ggplot_build(x)
 
-  if(is.null(g$plot$scales$scales[[1]]$palette.cache)) {
-    stop("You must choose options for the colour/fill in your ggplot2 object.")
-  }
+  ##Check which of the scales contains the colours
+  proto <- c()
+    for(i in 1:length(g$plot$scales$scales)){
 
-  if(is.null(base::names(g$plot$scales$scales[[1]]$palette.cache))) {
-    stop("You must provide names to the vector containing the colours you chose
-         which correspond to the group which is that colour:
-         scale_colour_manual(values = c('group 1' = 'colour 1', ...))")
-  }
+    if(!is.null(g$plot$scales$scales[[i]]$palette.cache)){
 
-  if(is.null(base::names(g$plot$scales$scales[[1]]$palette.cache))) {
-    stop("You must provide names to the vector containing the colours you chose
-         which correspond to the group which is that colour:
-         scale_colour_manual(values = c('group 1' = 'colour 1', ...))")
+      proto <- c(proto, i)
+
```

Comment:
could add a break clause here. No need to continue loop once proto found unless I'm missing something here

Reply:
Agreed, added